risc0/zkvm/src/receipt.rs

```diff
@@ -13,6 +13,55 @@
 // limitations under the License.
 
 //! Manages the output and cryptographic data for a proven computation
+//!
+//! # Usage
+//! Receipts are typically constructed as the output of the prover
+//! ```ignore
+//! // In real code, the ELF & Image ID would be generated by risc0 build scripts from guest code
+//! use methods::{EXAMPLE_ELF, EXAMPLE_ID};
+//! use risc0_zkvm::Prover;
+//!
+//! let mut prover = Prover::new(&EXAMPLE_ELF, EXAMPLE_ID)
+//!     .expect("Prover should be constructed from an ELF file and matching Image ID");
+//! let receipt = prover
+//!     .run()
+//!     .expect("Prover should run guest code that doesn't crash to generate a receipt");
+//! ```
+//! Once you have a Receipt, you can use the verify function to
+//! cryptographically verify that it was generated faithfully and is associated
+//! with the expected image ID. You can also read the contents of the journal.
+//! ```
+//! use risc0_zkvm::Receipt;
+//!
+//! # // Need to awkwardly set up a fake Receipt since we can't use the guest in docs
+//! # let journal_words = risc0_zkvm::serde::to_vec(&String::from("test"))
+//! #        .unwrap()
+//! #        .iter()
+//! #        .map(|&x| x.to_le_bytes())
+//! #        .collect::<Vec<[u8; 4]>>();
+//! # let mut journal: Vec<u8> = vec!();
+//! # for word in journal_words.iter() {
+//! #    for byte in word.iter() {
+//! #        journal.push(*byte);
+//! #    }
+//! # }
+//! # let receipt = Receipt {
+//! #    seal: vec!(),
+//! #    journal: journal,
+//! # };
+//! # use crate::risc0_zkvm::sha::Sha256;
+//! # let IMAGE_ID: [u32; 8] = [0, 0, 0, 0, 0, 0, 0, 0];
+//! // Here `receipt` is a Receipt whose journal contains the String "test"
+//! receipt.verify(&IMAGE_ID);
+//! let committed_value: String = risc0_zkvm::serde::from_slice(&receipt.journal)
+//!     .expect("Deserialization should succeed if the types match the committed types");
+//! assert_eq!("test", committed_value);
+//! ```
+//! Directly using the raw bytes of the journal is not recommended. Instead, we
```

Comment:
I don't think we should make recommendations, rather we should point out the pros/cons of various approaches.